DVCodingStyle.md

```diff
@@ -594,30 +594,122 @@ cause race conditions if there are any parallel threads.
 
 ### Objections and Coordinating End of Test
 
-1.  At a high level, a phase should not end when either there is more stimulus
-    to send during that phase or the DUT has yet to respond to some stimulus that
-    has been sent.
-    To prevent a phase from ending in the former case, raise and lower objections
-    inside tests.
-    To prevent a phase from ending in the latter case, raise objections in the
-    scoreboard whenever expect queues are not empty and lower them when they are empty.
-2.  It is recommended to avoid using objections in drivers or monitors
-    for performance reasons they should be used inside these components
-    sparingly and only when needed.
-    (Note: this point is still under discussion, and is still TBD)
-3.  If a sequence needs to raise/drop objections, it should call
-    `raise_objection` in its `pre_start()` method and `drop_objection` in its
-    `post_start()` method. A good alternative to this is to call
-    `set_automatic_phase_objection(1)` in the sequence constructor.
-4.  Always perform the test `if (get_starting_phase() != null)` before calling
-    `raise_objection` or `drop_objection` within a sequence.
-5.  When starting a sequence that can raise and drop objections, if you want
-    the sequence to raise and drop objections, set the `starting_phase` member of
-    the sequence object the `<sequence_handle>.set_starting_phase(<uvm_phase>)`
-    before starting the sequence.
-6.  To avoid reimplementing objection functionality in multiple sequence classes,
-    create a parent sequence that implements this functionality for its derivatives.
-7.  When calling `raise_objection` or `drop_objection`, you must pass a string
+
+At a high level, a phase should not end when either there is more stimulus to
+send during that phase or the DUT has yet to respond to some stimulus that has
+been sent.
+
+To prevent a phase from ending in the former case, raise and lower objections
+inside the test class.
+
+To prevent a phase from ending in the latter case, it is required to raise and
+lower objections inside the `phase_ready_to_end()` function.
```

Comment:
I think this should be "method", not "function". And probably this needs combining with the following paragraph (so that you can say whose method).

Reply:
You're probably right, I'll change this to 'method' instead of 'function'.

RE your second point - the general intent was to specify two separate recommendations, 1) that all objections should only be touched inside of `phase_ready_to_end()` and 2) we generally recommend objections should only be touched in the monitor.